src/lib/core/ipc.js

```diff
@@ -1,17 +1,31 @@
 const fs = require('./fs.js');
 const ipc = require('node-ipc');
+const os = require('os');
 const {parse, stringify} = require('flatted/cjs');
+const utils = require('../utils/utils.js');
 
 class IPC {
-
   constructor(options) {
     this.logger = options.logger;
-    this.socketPath = options.socketPath || fs.dappPath(".embark/embark.ipc");
+    this.socketPath = options.socketPath || IPC.ipcPath('embark.ipc');
     this.ipcRole = options.ipcRole;
     ipc.config.silent = true;
     this.connected = false;
   }
 
+  static ipcPath(basename, usePipePathOnWindows = false) {
```

Comment:
Personally, I would prefer to see `basename` and `usePipePathOnWindows` passed in to the constructor, and have `ipcPath` as a `get` property of the `IPC` class with a backing variable, so that this does not have to get calculated every time it is called.

This means that line 82 of `gethClient.js` would change to:
```
cmd.push(`--ipcpath=${new IPC({basename: 'geth.ipc', usePipePathOnWindows: true}).ipcPath}`);
```
and line 56 in `IPC.js` would change to:
```
fs.mkdirpSync(utils.dirname(this.ipcPath));
```

Although having to new an instance for simple property get does seem a bit wasteful. I guess this is the sort of thing that Dependency Injection is good at solving...

Reply:
Originally, I wanted to put `ipcPath()` as a helper in `core/fs.js`, as a sibling of `fs.dappPath()` and `fs.embarkPath()`. But I ran into a circular dependency problem, which was also the motivation for removing the `utils/utils.js` dependency in `core/env.js`. It may not be a problem anymore since `utils/host.ts` no longer needs to make use of `dappPath`, but I'm not 100% sure.

I'm trying to think if there would be any downsides to instantiating `IPC` in those additional places.

Reply:
@emizzle thanks for helping me rethink this a bit! The circular dep problem is gone after dropping `isDappMountedFromWindowsDockerHost` (forced pushed into oblivion a couple of commits ago).

I've now decreased the surface area of this PR, removing the changes to `core/env.js` and relocating the `ipcPath()` helper to `core/fs.js`, which seems a better place.

Thoughts?